### ISSUE TITLE
Relational: Entity splitting support

### ISSUE THREAD

**Issue by divega** [dotnet/efcore#620]
This work item represents the mapping capability to slice the data contained by an object into multiple tables.

Limitations:

- Properties mapped to multiple columns can't use store-generated values
- Foreign keys and indexes split across tables won't be created in the database

Related: https://github.com/dotnet/efcore/issues/17270#issuecomment-588037768

**Comment by AndriySvyryd** [comment]
Related to TPT https://github.com/aspnet/EntityFrameworkCore/issues/2266

**Comment by ajcvickers** [comment]
Note: consider optional relationships: #9773

**Comment by AndriySvyryd** [comment]
The implementation could be simplified if we used a database model #8258

**Comment by netspark** [comment]
Would be very appreciated for implementing this, we have a lot of legacy one-to-one tables in DB which |make no sense to treat separately in backend

**Comment by jansomers** [comment]
Would very much love this feature, also the optional relationships is a must have when implementing entity splitting

**Comment by maurei** [comment]
Any update? Would love to see this feature incorporated.

**Comment by ajcvickers** [comment]
@maurei This issue is in the Backlog milestone. This means that it is not going to happen for the 3.0 release. We will re-assess the backlog following the 3.0 release and consider this item at that time. However, keep in mind that there are many other high priority features with which it will be competing for resources.

**Comment by GeorgeTsaplin** [comment]
I suppose what [Owned Types](https://docs.microsoft.com/en-us/ef/core/modeling/owned-entities) can be used as an alternative for Entity Splitting and it's already implemented at EFCore 2.0.

**Comment by ArtemTereshkovich** [comment]
Hi. Do you have any update?

**Comment by mister-conor** [comment]
This feature is a must for people migrating legacy systems to EF.

**Comment by leus** [comment]
Yes please - consider the scenario where a table is split for patitionin causes (for example, to separate LOB and row data).

**Comment by bairog** [comment]
Lacking this feature in EF Core stops me (like many other in this thread) from migrating legacy projects from EF 6.4.
Do you have any concrete plan/schedule? What does "considered-for-net-release" mean? .NET 5 (scheduled in November 2020) or .NET Core 3.x (may happend earlier)?

**Comment by AndriySvyryd** [comment]
@bairog We'll consider adding it to .NET 5 if there's enough time left, otherwise it will likely go into .NET 7

**Comment by bairog** [comment]
Hello. Do you have any updates? Is it going to be included to EF Core 5 release?

**Comment by roji** [comment]
@bairog no, this feature did not make it into the upcoming 5 release. We'll consider this again for the 6.0 planning

**Comment by bairog** [comment]
Hello.
Are you planning to implement this in EF Core 6.0? It is vital for EF6 feature parity.
No single word about it in [this devblogs announcement](https://devblogs.microsoft.com/dotnet/the-plan-for-entity-framework-core-6-0/)

**Comment by ajcvickers** [comment]
@bairog This is a stretch goal for EF Core 6.0, as indicated by `consider-for-current-release` label. This means we may get to it for 6.0 if we have time in the schedule, but you shouldn't count on it.

**Comment by bairog** [comment]
So it looks like even EFCore 6.0 most likely will not have the entire list of current EF 6.4 features. Very disappointing to hear :(

**Comment by bairog** [comment]
EF Core 6 Preview 1 i[s out ](https://devblogs.microsoft.com/dotnet/announcing-entity-framework-core-6-0-preview-1/) - no word about Entity Splitting.
We've succesfully migrated most of our projects from EF 6.4.4 + Devart.Data.SQLite.EF6 provider (commercial) to EF Core 5 + Microsoft.EntityFrameworkCore.SQLite provider (free). Only one project (that uses Entity Splitting) cannot be ported because it uses Entity Splitting. Still hoping for EF Core 6,,

**Comment by bairog** [comment]
Hello again. It is definitely planned to be implemented in EF Core 7?
We've succesfully migrated most of our projects from EF 6.4.4 + Devart.Data.SQLite.EF6 provider (commercial) to EF Core 6 + Microsoft.EntityFrameworkCore.SQLite provider (free). Only one project cannot be ported because it uses Entity Splitting.

**Comment by ajcvickers** [comment]
@bairog While nothing in ~~life~~software is ever certain, I'm very confident that this will not be punted and will be included in EF7.

**Comment by DM2489** [comment]
I've been following this thread since EF Core 1.0.

We've also got a couple of apps that we've been forced to update to .NET Core (due to various other dependencies), and that have lost entity splitting capabilities.

We've had to come up with some weird, whacky, and nasty 'workarounds' to get over the lack of the entity splitting feature, such as combining multiple database tables into one, causing data to be replicated, etc.

I'd love to finally get my hands on this feature and tidy up these projects/databases to how they should be. I love .NET Core and EF Core but this single feature is the biggest thorn in my entire software stack at the moment - lack of entity splitting makes me feel as though I am intentionally developing antipattern applications, and there's not much I can really do about it without this feature 😨
